### ISSUE TITLE
[Feature request] Doubleclicking tags on "All tags" list should add them to "Image tags"

### ISSUE THREAD

**Issue by neojam** [starik222/BooruDatasetTagManager#4]
Hallo,

i would like to request a few enhancement features, if possible:

**1.**
Double-clicking tags on "All tags" list should add them to "Image tags".

Preferably with an option to select multiple images with CTRL+Leftclick or SHIFT+Leftclick (for from-to-selection) first and then add  tags to selected images by double-clicking on tags in "All tags" list.

With Addition to double-clicking, there should be dedicated buttons for it as well.
Like "add selected tags to selected images" and "Remove selected tags from selected images". 
Which would allow to select few images first, then select few tags from "all tags" list and finally mass-insert/mass remove those tags from selected images.

**2.**
When you filter images for specific tag with "Find in Dataset" button, the "Add to All " should only apply to filtered images. All other images should stay unaffected.

Would be great, if you could filter with "Find in Dataset" first, then select several tags from "All tags" list and then either add or remove all selected tags from the filtered image-list. Other images should stay unaffected. 
There still should be another button to replace tags in ALL images. So 4 buttons:
Add to filtered list 
Remove from filtered list
Add to all
Remove from all

**3.**
De-duplication of tags is needed.
Currently its possible to have the same tag multiple times for the same image.
Would be great if BooruDatasetTagManager would check and only leave first instance of the tag in a txt-File and remove all other dups of that tag from the list.

The First dedup process to be ran at "File>Open Folder".
The second at "File>Save all changes".

**4.**
If there is no image file for a .txt file, the BooruDatasetTagManager should display a placeholder/"No Image" thumbnail.
Currently there is an application error, if you just want to edit txt descriptions files without images.

**5.** 
Sorting Options for thumbnails are needed (currently the thumbs are displayed in seemingly some random order).
Sort thumbnails by:
- Alphabetically by Filename
- By modification date of txt files
- by modification date of img files

thank you!

**Comment by starik222** [comment]
Ok, i will try to implement this

**Comment by neojam** [comment]
Thank you :)


P.S.
Can you tell me how the preview thumbnails of the "Dataset" list get sorted?
I need them to be sorted alphabetically, based on the filenames:
a001.txt/jpg, 
a002.txt/jpg, 
b001.txt/jpg, 
b002.txt/jpg,
But currently it appears that the thumbs are sorted in a  random order...

Any way to sort them differently? 
If not, then its another feature request, i guess ^^'

**Comment by starik222** [comment]
> If there is no image file for a .txt file, the BooruDatasetTagManager should display a placeholder/"No Image" thumbnail.
Currently there is an application error, if you just want to edit txt descriptions files without images.

I do not quite understand how you get this error, do you delete images while working in the program? Loading occurs by images, therefore, even if there are text files without images in the folder, they will not be loaded.

**Comment by neojam** [comment]
> > If there is no image file for a .txt file, the BooruDatasetTagManager should display a placeholder/"No Image" thumbnail.
> > Currently there is an application error, if you just want to edit txt descriptions files without images.
> 
> I do not quite understand how you get this error, do you delete images while working in the program? Loading occurs by images, therefore, even if there are text files without images in the folder, they will not be loaded.

I tried to open a folder containing only txt files without images. Thats when the error happened.
Reason for it:
I have a dataset of 300 Images and txt files.
In that dataset I have 3 different characters name tags. There are a few images of character A, then there are a few images of B and a few of C. I need to add a few character specific tags for all the characters separately.
To do that:
I grepped the **contents** of all txt files for A and found it in 80 txt files. 
Then i moved those txt files into a folder (just simple grep and move commands). Then I just wanted to add a few tags with BooruDatasetTagManager to those txt files, which is when i got the error.

Since BooruDatasetTagManager can't apply tags to filtered images (my Feature request "2." on the list), I have to use this workaround (i now moving images with txt files, for this workaround to work.)

But Ideally BooruDatasetTagManager should be able to open txt files without the images too.
